File: python-app/models/pivot_table/pivot_table_level.py
from enum import Enum

class PivotTableLevel(Enum):
    GROUP = "group"
    YEAR = "year"
    SUBJECT = "subject"
    PROFESSOR = "professor"
    UNIT = "unit"
    GRADE_TYPE = "gradeType"

def level_to_spanish(level: PivotTableLevel):
    match level:
        case PivotTableLevel.GROUP:
            return "Grupo"
        case PivotTableLevel.SUBJECT:
            return "Materia"
        case PivotTableLevel.PROFESSOR:
            return "Profesor"
        case PivotTableLevel.UNIT:
            return "Unidad"
        case PivotTableLevel.GRADE_TYPE:
            return "Tipo de calificación"
        case _:
            raise ValueError(f"Nivel desconocido: {level}")

File: lib/models/pivot_table/pivot_table_level.dart
enum PivotTableLevel { group, year, subject, professor, unit }

String levelToSpanish(PivotTableLevel level) {
  switch (level) {
    case PivotTableLevel.group:
      return "Grupo";
    case PivotTableLevel.professor:
      return "Profesor";
    case PivotTableLevel.subject:
      return "Materia";
    case PivotTableLevel.unit:
      return "Unidad";
    case PivotTableLevel.year:
      return "Año";
  }
}
